src/uwtools/tests/config/test_validator.py

```diff
@@ -202,6 +213,35 @@ def test_validate_yaml_rocoto_invalid_walltime_pattern(rocoto_assets, caplog):
         assert logged(caplog, "'0:01:00' does not match '^[0-9]{2}:[0-9]{2}:[0-9]{2}$'")
 
 
+def test_validate_yaml_rocoto_invalid_dependency_bool(rocoto_assets, caplog):
+    kwargs, config = rocoto_assets
+    config["workflow"]["tasks"]["metatask"]["task"]["dependency"].update(
+        {"maybe": {"taskdep": {"attrs": {"task": "hello"}}}}
+    )
+    with patch.object(validator, "YAMLConfig") as YAMLConfig:
+        YAMLConfig().data = config
+        assert not validator.validate_yaml(**kwargs)
+        assert logged(
+            caplog,
+            "'maybe' does not match any of the regexes: "
+            "'^(and|or|not|nand|nor|xor)(_.*)?$', '^(streq|strneq)(_.*)?$', "
+            "'^cycleexistdep(_.*)?$', '^datadep(_.*)?$', '^metataskdep(_.*)?$', "
+            "'^some(_.*)?$', '^taskdep(_.*)?$', '^taskvalid(_.*)?$', '^timedep(_.*)?$'",
+        )
```

Comment:
I really don't like this long line, but I had issues getting the first part to match I think due to the `records` in `logged()`

Reply:
I can give you a trick for doing a partial match, which `logged()` doesn't support. I'll get back to you shortly with it...

Reply:
In `test_templater.py`, there's currently this assertion using `logged()`:
```
assert logged(caplog, "roses are red, violets are blue")
```
That has to match, as you noted, a complete line of log output. If I do `import re` at the top of the test module, I could alternatively do things like
```
assert any(re.match(r"^.*red, violets.*$", record.message) for record in caplog.records)
```
to assert that `red, violets` appears on some line, anywhere on that line or, say,
```
assert any(re.match(r"^roses are red.*$", record.message) for record in caplog.records)
```
to assert that `roses are red` appears at the _start_ of some line.

If you find this useful, it might be worth adding it to `uwtools.tests.support` as a helper similar to `logged()` so that you could use it now but it would also be available for future tests without much repetition.

Reply:
Ah, ok - hmm, could we update `logged()` to something like `return any(msg in [record.message for record in caplog.records])`? We could also move the re.match there - I don't like the way the list comprehension looks personally, though

Reply:
In re: `return any(msg in [record.message for record in caplog.records])`, I don't think so, because the expression `msg in [record.message for record in caplog.records]` evaluates to a `bool` value, and `any()` accepts an iterable as its argument:
```
>>> any(2 in [1, 2, 3])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bool' object is not iterable
```
If we made `logged()` regex-based, all call sites would have to be updated to give complete, anchored regexes, e.g. `^roses are red, violets are blue$` instead of simple strings, to avoid unintentional partial matches. I feel like it's common enough that we can and want to match complete log lines that having a simple, non-regex based `logged()` is helpful and simple to use in most use cases; having something like `logged_regex()` for power-use cases might be nice to have, though -- your use case certainly begs for it.

Reply:
Hmm, yeah, I missed that - breaking it out of the brackets to just  `return any(msg in record.message for record in caplog.records)` works by leaving the `any()` to do the evaluation, but I think I'll still break the regex off into a helper like you said though